app/src/main/java/com/example/wnews/MainActivity.kt

```diff
@@ -2,10 +2,106 @@ package com.example.wnews
 
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import MainViewModel
+import android.content.Intent
+import androidx.lifecycle.ViewModelProvider
+import com.example.wnews.databinding.ActivityMainBinding
 
 class MainActivity : AppCompatActivity() {
+    private lateinit var mainViewModel : MainViewModel
+    private lateinit var binding : ActivityMainBinding
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        binding= ActivityMainBinding.inflate(layoutInflater)
+        val view = binding.root
+        setContentView(view)
+
+        mainViewModel = ViewModelProvider(this)[MainViewModel::class.java]
+
+        binding.apply {
+
+
+            loginButton.setOnClickListener{
+                println("user:"+ loadData())
+                val firstNameInput = firstName.editableText.toString()
+                val secondNameInput= secondName.editableText.toString()
+
+                mainViewModel.setLogin(firstNameInput, secondNameInput)
+
+                if(loginValidation(firstNameInput, secondNameInput)){
+                    // fun save userData()
+                    val intent = Intent(this@MainActivity, HomeActivity::class.java)
+
+                    startActivity(intent)
+                }
+            }
+            fun initHome(){
+                loginButton.setOnClickListener { mainViewModel.setLogin(firstName =String(), secondName =String()) }
+            }
+
+        }
+
+        setObserve()
+
+    }
+
+    fun loginValidation(firstNameInput: String, secondNameInput: String): Boolean{
+        if(firstNameInput.isEmpty()) {
+            binding.firstName.error = "This field cannot be empty."
+            return false
+        }
+
+        if (firstNameInput.isNotEmpty() && !firstNameInput.contains("@")) {
+            binding.firstName.error = "Please, enter a valid email."
+            return false
+        }
+
+        if(secondNameInput.isEmpty()) {
+            binding.secondName.error = "This field cannot be empty."
```

Comment:
Once we talk about leave plain text in the document. This kind of text, have to be in "string.xml" file. And then, you call it like this:
 binding.firstName.error = getString(R.string.error_empty)